smithy-rules-engine/src/main/java/software/amazon/smithy/rulesengine/traits/EndpointTestsTraitValidator.java

```diff
@@ -28,6 +39,28 @@
 public final class EndpointTestsTraitValidator extends AbstractValidator {
     @Override
     public List<ValidationEvent> validate(Model model) {
-        return null;
+
+        List<ValidationEvent> result = new ArrayList<>();
+        for (ServiceShape service : model.getServiceShapesWithTrait(EndpointTestsTrait.class)) {
+            result.addAll(
+                    service.expectTrait(EndpointTestsTrait.class).getTestCases().stream()
+                    .flatMap(endpointTestCase -> endpointTestCase.getOperationInputs().stream())
+                    .flatMap(operationInput -> validateOperationInput(operationInput, service, model))
+                    .collect(Collectors.toList()));
+        }
+        return result;
+    }
+
+    private Stream<ValidationEvent> validateOperationInput(EndpointTestOperationInput testOperationInput, ServiceShape serviceShape, Model model) {
+        OperationIndex index = OperationIndex.of(model);
+        StructureShape input = index.expectInputShape(ShapeId.from(testOperationInput.getOperationName()));
```

Comment:
Are you planning separate changes to validate the buildIn/clientParams (at least against the ruleset.  Both for required/type?)